src/dataspec/base.py

```diff
@@ -150,77 +150,144 @@ def tag(self) -> Tag:  # pragma: no cover
-        False."""
+        """
+        Returns :py:obj:`True` if ``v`` is valid according to the Spec, otherwise
+        returns :py:obj:`False` .
+
+        :param v: a value to validate
+        :return: :py:obj:`True` if the value is valid according to the Spec, otherwise
+            :py:obj:`False`
+        """
         try:
             next(self.validate(v))
         except StopIteration:
             return True
         else:
             return False
 
-    def _default_conform(self, v):
+    @property
+    def conformer(self) -> Optional[Conformer]:
+        """Return the custom conformer attached to this Spec, if one is defined."""
+        return None
+
+    def conform(self, v: Any):
         """
-        Default conformer for the Spec.
+        Conform ``v`` to the Spec, returning the possibly conformed value or an
+        instance of :py:class:`dataspec.Invalid` if the value cannot be conformed.
 
-        If no custom conformer is specified, this conformer will be invoked.
+        :param v: a value to conform
+        :return: a conformed value or a :py:class:`dataspec.Invalid` instance if the
+            input value could not be conformed
         """
-        return v
+        if self.is_valid(v):
+            return self.conform_valid(v)
+        else:
+            return INVALID
 
-    def conform_valid(self, v):
+    def conform_valid(self, v: Any):
         """
         Conform ``v`` to the Spec without checking if v is valid first and return the
         possibly conformed value or ``INVALID`` if the value cannot be conformed.
 
         This function should be used only if ``v`` has already been check for validity.
-        """
-        return self.conformer(v)
 
-    def conform(self, v: Any):
-        """Conform ``v`` to the Spec, first checking if v is valid, and return the
-        possibly conformed value or ``INVALID`` if the value cannot be conformed."""
-        if self.is_valid(v):
-            return self.conform_valid(v)
-        else:
+        :param v: a *validated* value to conform
+        :return: a conformed value or a :py:class:`dataspec.Invalid` instance if the
+            input value could not be conformed
+        """
+        if self.conformer is None:
+            return v
+        try:
+            return self.conformer(v)  # pylint: disable=not-callable
+        except Exception:
             return INVALID
 
-    def with_conformer(self, conformer: Conformer) -> "Spec":
-        """Return a new Spec instance with the new conformer."""
+    def compose_conformer(self, conformer: Conformer) -> "Spec":
+        """
+        Return a new Spec instance with a new conformer which is the composition of the
+        ``conformer`` and the current conformer for this Spec instance.
+
+        If the current Spec instance has a custom conformer, this is equivalent to
+        calling ``spec.with_conformer(lambda v: new_conformer(spec.conformer(v)))`` .
```

Comment:
I this this makes it clearer that the `conformer` passed to the method is used:
```suggestion
        calling ``spec.with_conformer(lambda v: conformer(spec.conformer(v)))`` .
```